### PR TITLE
Fix: Publish Beta & Dev Release - Core [ED-10488]

### PR DESCRIPTION
**Pull request by maork-elementor**

### CODE REVIEW

File: .github/workflows/publish-v2.yml [hunk 1]
```diff
@@ -86,14 +86,35 @@ jobs:
...
+        if: env.CHANNEL != 'cloud' && github.event.inputs.pre_release == 'true'
+        env:
+          PLUGIN_VERSION: ${{ env.RELEASE_NAME }}
+          SVN_PASSWORD: ${{ secrets.SVN_PASSWORD }}
+          SVN_USERNAME: ${{ secrets.SVN_USERNAME }}
+          CHANNEL: ${{ env.CHANNEL }}
+          PACKAGE_VERSION : ${{ env.PACKAGE_VERSION }}
+        run: |
+          unzip -q /tmp/${{ github.event.repository.name }}-${{ env.RELEASE_FILENAME }}.zip -d $GITHUB_WORKSPACE
+          bash "${GITHUB_WORKSPACE}/.github/scripts/validate-build-files.sh"
```
Comment (yotamselementor) [inline .github/workflows/publish-v2.yml:99]: Don't we want to run the validation also on regular releases?

**Review by yotamselementor — CHANGES_REQUESTED** [review]

File: .github/scripts/validate-build-files.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+fi
+
+if [[ $(grep -c "Version: $PLUGIN_VERSION" elementor.php) -eq 0 ]]; then
+	echo "elementor.php file does not contain the correct version"
+	exit 1
+fi
+
+if [[ "$CHANNEL" == "ga" ]]; then
+	if [[ $(grep -c "Stable tag: $PACKAGE_VERSION" readme.txt) -eq 0 ]]; then
+		echo "readme.txt file does not contain the correct version"
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/scripts/validate-build-files.sh:47]: Done

File: .github/scripts/validate-build-files.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		exit 1
+	fi
+fi
+
+if [[ "$CHANNEL" == "beta" ]]; then
+	if [[ $(grep -c "Beta tag: $PLUGIN_VERSION" readme.txt) -eq 0 ]]; then
+		echo "readme.txt file does not contain the correct version"
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/scripts/validate-build-files.sh:54]: Done

File: .github/workflows/get-release-name/action.yml [hunk 4]
```diff
@@ -28,10 +28,9 @@ runs:
             # e.g. elementor-3.11.0-cloud20 => v3.11.0-cloud20
             RELEASE_NAME=${RELEASE_NAME/${{ inputs.PLUGIN_NAME }}-/v}
 
-            # e.g. 3.11.0 => v3.11.0
-            if [[ ! $RELEASE_NAME =~ ^v ]]; then
-              RELEASE_NAME=v$RELEASE_NAME
-            fi
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/get-release-name/action.yml:34]: This `v` is coming from the `create-rc-tag` action.
The RC tag name is containing the `v` prefix.

Currently, I believe we shouldn't make any changes to this file so we won't introduce new bugs.

File: .github/workflows/publish-v2.yml [hunk 5]
```diff
@@ -86,14 +86,35 @@ jobs:
           environment-id: "${{ env.PACKAGE_VERSION }}"
           environment-display-name: "${{ env.PACKAGE_VERSION }}"
           environment-type: 'production'
+      - name: Debug Publish to WordPress.org SVN
+        if: env.CHANNEL != 'cloud' && github.event.inputs.pre_release == 'true'
+        env:
+          PLUGIN_VERSION: ${{ env.RELEASE_NAME }}
+          SVN_PASSWORD: ${{ secrets.SVN_PASSWORD }}
+          SVN_USERNAME: ${{ secrets.SVN_USERNAME }}
```
[the remark this answers is not in the document]
Comment (maork-elementor) [reply .github/workflows/publish-v2.yml:94]: Done

File: .github/workflows/publish-v2.yml (hunk 1 above)
Comment (maork-elementor) [reply .github/workflows/publish-v2.yml:99]: As you can see in the link, we do validate also on regular releases: 

https://github.com/maork-elementor/elementor/blob/368839e3caaf30594d7922b08855b0be0a600b48/.github/workflows/publish-v2.yml#L117

**Review by yotamselementor — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-v2.yml (hunk 1 above)
Comment (maork-elementor) [reply .github/workflows/publish-v2.yml:97]: Done

**Review by yotamselementor — APPROVED** [review]

**Pull request merged**